src/main/java/edu/kis/powp/jobs2d/command/manager/XMLCommandLoader.java

```diff
@@ -0,0 +1,78 @@
+package edu.kis.powp.jobs2d.command.manager;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import org.w3c.dom.*;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+
+public class XMLCommandLoader implements LoaderCommand {
+    private String path, name;
+
+    public XMLCommandLoader(String path, String name) {
+        this.path = path;
+        this.name = name;
+    }
+
+    @Override
+    public synchronized LoadedCommand loadCommandFromExternalSource() {
+        File file = new File(path);
+        DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder dBuilder;
+        List<Command> commandsList = new ArrayList<>();
+        try {
+            dBuilder = dbFactory.newDocumentBuilder();
+            Document doc = dBuilder.parse(file);
```

Comment:
if possible push file system dependency to the GUI class. Can builder work with a string?

Reply:
Not with string, but i can work with InputStream that can be created from String. Changed it to work this way. Done.